anvil/tests/it/transaction.rs

```diff
@@ -304,3 +315,40 @@ contract Contract {
         assert!(err.to_string().contains("execution reverted:"));
     });
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn get_past_events() {
+    let (_api, handle) = spawn(NodeConfig::test().with_port(next_port())).await;
+    let provider = handle.http_provider();
+
+    let wallet = handle.dev_wallets().next().unwrap();
+    let address = wallet.address();
+    let client = Arc::new(SignerMiddleware::new(provider, wallet));
+
+    let contract = SimpleStorage::deploy(Arc::clone(&client), "initial value".to_string())
+        .unwrap()
+        .send()
+        .await
+        .unwrap();
+
+    let func = contract.method::<_, H256>("setValue", "hi".to_owned()).unwrap();
+    let tx = func.send().await.unwrap();
+    let _receipt = tx.await.unwrap();
+
+    // and we can fetch the events
+    let logs: Vec<ValueChanged> =
+        contract.event().from_block(0u64).topic1(address).query().await.unwrap();
+
+    // 2 events, 1 in constructor, 1 in call
+    assert_eq!(logs[0].new_value, "initial value");
+    assert_eq!(logs[1].new_value, "hi");
+    assert_eq!(logs.len(), 2);
+
+    // and we can fetch the events at a block hash
+    let hash = client.get_block(1).await.unwrap().unwrap().hash.unwrap();
+
+    let logs: Vec<ValueChanged> =
```

Comment:
Nitpicking, but could be named `logs_at_block_1` to make the assertions at line 352 and 353 clearer and to not have two vars with the same name